coderd/healthcheck/derphealth/derp.go

```diff
@@ -24,9 +24,14 @@ import (
 	"github.com/coder/coder/v2/coderd/util/ptr"
 )
 
+const (
+	warningNodeUsesWebsocket = `Node uses WebSockets because the "Upgrade: DERP" header may be blocked on the load balancer.`
+)
+
 // @typescript-generate Report
 type Report struct {
-	Healthy bool `json:"healthy"`
+	Healthy  bool     `json:"healthy"`
+	Warnings []string `json:"warnings"`
```

Comment:
Should this be a `map[string][]string` so that it's easier to identify the source of the warnings?

That way we don't need to worry about doing `fmt.Sprintf("[%s] %s", prefix, warning")` all the time.

Reply:
It might also make easier for the UI to display them separately.

Reply:
> It might also make easier for the UI to display them separately.

My main concern is consistency with other fields, but I don't mind changing it.

Reply:
I'm not sure what would be easier for the UI, to be honest. Maybe we just ship and iterate?

Reply:
Agree 👍